src/Analyzers/MSTest.GlobalConfigsGenerator/AnalyzerSeverityDecider.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.CodeAnalysis;
+
+// NOTE for our current matrix.
+// We decide whether to include a rule in a specific mode or not based on two factors.
+// 1. IsEnabledByDefault (true/false)
+// 2. DefaultSeverity (Info, Warn) - note: we currently don't use severity Hidden or Error.
+// So, we have 4 possible combinations.
+// 1. IsEnabledByDefault = true, DefaultSeverity = Warn (we include that in All, Recommended, and Default)
+// 2. IsEnabledByDefault = true, DefaultSeverity = Info (we include that in All and Recommended)
+// 3. IsEnabledByDefault = false, DefaultSeverity = Warn (we don't combine Warn with not enabled by default)
+// 4. IsEnabledByDefault = false, DefaultSeverity = Info (we include that in All only)
+internal static class AnalyzerSeverityDecider
+{
+    public static DiagnosticSeverity? GetSeverity(DiagnosticDescriptor rule, MSTestAnalysisMode mode)
+    {
+        DiagnosticSeverity? severity = mode switch
+        {
+            MSTestAnalysisMode.All => DecideForModeAll(rule),
+            MSTestAnalysisMode.Recommended => DecideForModeRecommended(rule),
+            MSTestAnalysisMode.Default => DecideForModeDefault(rule),
+            MSTestAnalysisMode.None => DecideForModeNone(rule),
+            _ => throw new ArgumentException($"Unexpected MSTestAnalysisMode '{mode}'.", nameof(mode)),
+        };
+
+        return severity;
+    }
+
+    private static DiagnosticSeverity? DecideForModeAll(DiagnosticDescriptor rule) =>
+        // TODO: We should consider at least not enabling "conflicting" rules.
```

Comment:
We should address this, I think.

Reply:
Yes I agree. Shall we have some extra tags for such rules?

Reply:
@Evangelink We can either implement it "in place" inside the AnalyzerSeverityDecider or have an extra tag. I'm not sure why roslyn-analyzers went with extra tags but implementing in-place is a bit easier for sure. Let's discuss later.